datahub-web-react/src/app/onboarding/utils.tsx

```diff
@@ -15,6 +15,43 @@ export function getStepIds(userUrn: string) {
     return OnboardingConfig.map((step) => `${userUrn}-${step.id}`);
 }
 
+// if the user just saw the preRequisiteStepId (in stepIdsToAdd) of a conditional step, we need to add this conditional step
+export function getConditionalStepIdsToAdd(providedStepIds: string[], stepIdsToAdd: string[]) {
+    const conditionalStepIds: string[] = [];
+
+    const providedSteps = providedStepIds
+        .map((stepId) => OnboardingConfig.find((step: OnboardingStep) => step.id === stepId))
+        .filter((step) => !!step);
+
+    providedSteps.forEach((step) => {
+        if (
+            step?.preRequisiteStepId &&
+            stepIdsToAdd.includes(step?.preRequisiteStepId) &&
+            !stepIdsToAdd.includes(step.id || '')
+        ) {
+            conditionalStepIds.push(step.id || '');
+        }
+    });
+
+    return conditionalStepIds;
+}
+
+function hasStepBeenSeen(stepId: string, userUrn: string, educationSteps: StepStateResult[]) {
+    const convertedStepId = convertStepId(stepId, userUrn);
+    return educationSteps.some((step) => step.id === convertedStepId);
+}
+
+// add conditional steps if they have seen the pre-requisite step only
+export function filterConditionalStep(step: OnboardingStep, userUrn: string, educationSteps: StepStateResult[]) {
```

Comment:
returns true if?

Reply:
better to name it in a way that describes the operation, similar to "hasStepBeenSeen".

Could this method could be named "hasX"?

Reply:
you got it - updating now